### PR TITLE
kie-issues#9: Transform OpenShift authentication to AuthProvider, enabling multiple OpenShift instances to be configured

### LINKED ISSUE #9: Transform OpenShift authentication to AuthProvider, enabling multiple OpenShift instances to be configured
On KIE Sandbox, it is possible to authenticate with an OpenShift instance, allowing for development deployments of DMN models.

This configuration is currently present on the Settings modal, so only one is supported.

By creating a new AuthProvider type "openshift", users can authenticate with many OpenShift instances, allowing for them to choose where to deploy their DMN models during development.

It is expected that the implications of this move will also be addressed on this task, like some sort of selector for users to choose where their DMN models will be deployed to. Also, the lists of deployments needs to be parameterized by the OpenShift instance choice.

### PR DESCRIPTION
**Pull request by tiagobento**
Closes https://github.com/kiegroup/kie-issues/issues/9

---

This PR introduces a few important changes regarding to authentication on KIE Sandbox. The Accounts modal is now used for Git and Cloud providers, allowing users to authenticate and deploy models to more than one OpenShift instance. The authentication steps are pretty much the same as before, they're just in another place, and support multiple accounts to be stored in KIE Sandbox.

The `DMN Dev Sandbox` feature has been renamed to `DMN Dev deployments`, part of a larger feature called simply `Dev deployments`. The OpenShift deployments dropdown changed to "Dev deployments" as well, allowing for other Cloud providers to be added in the future, like plain Kubernetes.

A few changes have been made to the dropdown itself, allowing users to select which AuthSession they want to use to see their deployments. Same for the `Deploy` option for DMN models.

Right now, the Dev deployments feature is exclusive to DMNs only, but that can (and most likely will) change in the future.

@caponetto Can you please create the org on quay.io for the new DMN Dev deployment base image? 😄 

Also in this PR:
- Made some minor changes to `dmn-dev-deployment-form-webapp`
- Added the `group` property to AuthProviders.
- Improved AuthProviders icons with colors! 🎨
- Reorganized global contexts to fix a bug that was happening when trying to navigate away from a GitHub Workspace that contained changes. Needed to move `GlobalAlertsContextProvider` below a few levels.
- Enlarged the Accounts modal
- Changed some texts to not be so DMN-centric regarding Dev deployments
- Improved AuthSessionSelect filters
- Made some Dev deployments alerts close automatically after 5 seconds.
- Parameterized all OpenShift-related operations with an `OpenShiftConnection`.
- Removed old OpenShift Settings tab 
- Fixed a bug on `extended-services` to prevent concatenating HTTP response headers.
- Some adjustments on the Extended Services endpoints mechanism.
- Promoted `authSessions` and `authProviders` to the top-level directory, not under `accounts` anymore.
- Exposed a new method on `useLivePromiseState` to allow imperative refreshes.

---

# DEMO

1. The Accounts modal now supports filtering by "group" of AuthProvider, allowing Git-related AuthSessionSelect instances to direct users to Git providers, and Cloud-related AuthSessionSelect instances to direct users to Cloud providers.

https://user-images.githubusercontent.com/1584568/203658081-5d35b5c2-6afc-40c6-9654-8ccebad608b1.mov

2. Deleting Dev deployments continue to work normally, and the list of Deployments to be deleted appears on the confirmation modal.

https://user-images.githubusercontent.com/1584568/203658711-0949e401-6f1a-4ad4-8fdf-a7ae73c49e02.mov

3. Connecting to an OpenShift instance from the Dev deployments dropdown will open the Modal upon connection. 

https://user-images.githubusercontent.com/1584568/203658714-aa21dd92-ca17-4e7e-b693-d4c6ddd4acae.mov

4. Deploying DMN decisions works much in the same way as before too.

https://user-images.githubusercontent.com/1584568/203658950-f1bf8c4a-1796-4076-987f-17ec2e8c5c29.mov

### CODE REVIEW

**Comment by caponetto** [comment]
@tiagobento I just created https://quay.io/repository/kie-tools/dmn-dev-deployment-base-image

**Comment by tiagobento** [comment]
Just noticed this is missing an empty state for when there's no deployments. Please hold merging this. Reviewing is fine though, as the changes I need to make are minimal.

File: packages/dmn-dev-deployment-form-webapp/src/DmnFormToolbar.tsx [hunk 1]
```diff
@@ -90,6 +90,7 @@ export function DmnFormToolbar(props: Props) {
       >
         <Text className={"kogito--dmn-form__toolbar-title"}>
           {i18n.formToolbar.disclaimer.title}
+          &nbsp; &nbsp;
```
Comment (caponetto) [inline packages/dmn-dev-deployment-form-webapp/src/DmnFormToolbar.tsx:93]: At some point, using `pf-u-ml-sm` like in the line below stopped working on all packages 🤔

File: packages/dmn-dev-deployment-base-image/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "build:prod:win32": "echo \"Build not supported on Windows\"",
+    "cleanup": "rimraf dist-dev && mkdir dist-dev",
+    "copy:assets": "pnpm copy:dmn-dev-deployment-form-webapp && pnpm copy:dmn-dev-deployment-quarkus-app",
+    "copy:dmn-dev-deployment-form-webapp": "cp -R ../dmn-dev-deployment-form-webapp/dist ./dist-dev/dmn-dev-deployment-form-webapp",
+    "copy:dmn-dev-deployment-quarkus-app": "cp -R ../dmn-dev-deployment-quarkus-app ./dist-dev/dmn-dev-deployment-quarkus-app",
+    "podman:build": "run-script-if --bool $([ $(command -v podman) ] && echo true || echo false) --then \"podman build $(echo $(build-env dmnDevDeploymentBaseImage.buildTags) | xargs printf -- \"-t $(build-env dmnDevDeploymentBaseImageEnv.registry)/$(build-env dmnDevDeploymentBaseImageEnv.account)/$(build-env dmnDevDeploymentBaseImageEnv.name):%s\n\" | xargs echo) --build-arg QUARKUS_PLATFORM_VERSION=$(build-env quarkusPlatform.version) --build-arg KOGITO_RUNTIME_VERSION=$(build-env kogitoRuntime.version) -f Containerfile\" --else \"echo Podman not found, skipping image build.\""
```
Comment (caponetto) [inline packages/dmn-dev-deployment-base-image/package.json:24]: I really liked what you did for the KIE Sandbox image (add support for docker, which helps people a lot on macOS).
Consider doing the same for this image in the future. :)

**Review by caponetto — APPROVED** [review]

**Pull request merged**